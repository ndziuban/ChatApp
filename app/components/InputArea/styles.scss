.input__container {
  padding: 5px;
  
  .input {
    width: 100%;
    box-sizing: border-box;
    resize: none;
    padding: 10px 15px;
    border-radius: 20px;
    border: 1px solid #d7d7d7;
    font-size: 14px;
    box-shadow: 0 4px 6px rgba(0,0,0,0.1);
    background: #f7f7f7;

    &:focus {
      outline: none;
    }
  }
}
