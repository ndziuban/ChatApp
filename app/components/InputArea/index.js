import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

class Input extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { value: '' };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.onEnterPress = this.onEnterPress.bind(this);
  }

  handleChange(e) {
    this.setState({ value: e.target.value });
    this.props.onChange();
  }

  handleSubmit(e) {
    if(this.state.value.match(/\S/g)) {
      this.props.onSubmit(this.state.value)
      this.setState({ value: '' });
    }
    e.preventDefault();
  }

  onEnterPress(e) {
    if(e.keyCode == 13 && e.shiftKey == false) {
      e.preventDefault();
      this.handleSubmit();
    }
  }

  render() {
    return (
      <div>
        <textarea
          rows="2"
          className="input"
          placeholder="Type here to send a message..."
          value={this.state.value}
          onChange={this.handleChange}
          onKeyDown={this.onEnterPress}
        />
      </div>
    );
  }
};

Input.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Input;
