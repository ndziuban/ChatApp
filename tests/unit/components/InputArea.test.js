import React from 'react';
import InputArea from '../../../app/components/InputArea';
import renderer from 'react-test-renderer';

it('InputArea renders correctly', () => {
  const tree = renderer
    .create(<InputArea onChange={jest.fn()} onSubmit={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
