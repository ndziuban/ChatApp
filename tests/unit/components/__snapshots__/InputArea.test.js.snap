// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputArea renders correctly 1`] = `
<div
  className="input__container"
>
  <textarea
    className="input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type here to send a message..."
    rows="2"
    value=""
  />
</div>
`;
